packages/matter.js/src/protocol/interaction/InteractionServer.ts

```diff
@@ -1351,41 +1340,41 @@ export class InteractionServer implements ProtocolHandler<MatterDevice>, Interac
-                    async () =>
-                        await this.invokeCommand(
-                            path,
-                            command,
-                            exchange,
-                            commandFields ?? TlvNoArguments.encodeTlv(commandFields),
-                            message,
-                            endpoint,
-                            receivedWithinTimedInteraction,
-                        ),
-                    StatusResponseError,
-                    async error => {
-                        let errorCode = error.code;
-                        const errorLogText = `Error ${Diagnostic.hex(errorCode)}${
-                            error.clusterCode !== undefined ? `/${Diagnostic.hex(error.clusterCode)}` : ""
-                        } while invoking command: ${error.message}`;
-                        if (error instanceof ValidationError) {
-                            logger.info(
-                                `Validation-${errorLogText}${error.fieldName !== undefined ? ` in field ${error.fieldName}` : ""}`,
-                            );
-                            if (errorCode === StatusCode.InvalidAction) {
-                                errorCode = StatusCode.InvalidCommand;
-                            }
-                        } else {
-                            logger.info(errorLogText);
+                let result;
+                try {
+                    result = await this.invokeCommand(
+                        path,
+                        command,
+                        exchange,
+                        commandFields ?? TlvNoArguments.encodeTlv(commandFields),
+                        message,
+                        endpoint,
+                        receivedWithinTimedInteraction,
+                    );
+                } catch (e) {
+                    StatusResponseError.accept(e);
+
+                    let errorCode = e.code;
+                    const errorLogText = `Error ${Diagnostic.hex(errorCode)}${
+                        e.clusterCode !== undefined ? `/${Diagnostic.hex(e.clusterCode)}` : ""
+                    } while invoking command: ${e.message}`;
+
+                    if (e instanceof ValidationError) {
+                        logger.info(
+                            `Validation-${errorLogText}${e.fieldName !== undefined ? ` in field ${e.fieldName}` : ""}`,
+                        );
+                        if (errorCode === StatusCode.InvalidAction) {
+                            errorCode = StatusCode.InvalidCommand;
                         }
-                        return {
-                            code: errorCode,
-                            clusterCode: error.clusterCode,
-                            responseId: command.responseId,
-                            response: TlvNoResponse.encodeTlv(),
-                        };
-                    },
-                );
+                    }
```

Comment:
else clause missing that logs error text as info

Reply:
fixed